### PR TITLE
refactor: migrate multiple application api call to Redux level.

### PR DESCRIPTION
**Pull request by aryanshridhar**
Added a pr to migrate multiple Application api calls to Redux.

#### To-Do

- [x] Request per second api call.
- [x] Error Percentage api call.

### CODE REVIEW

File: frontend/src/api/metrics/getErrorPercentage.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import axios from 'api';
+import { ErrorResponseHandler } from 'api/ErrorResponseHandler';
```
Comment (palash-signoz) [inline frontend/src/api/metrics/getErrorPercentage.ts:2]: can we use single api as we are just passing query in each API

Comment (palash-signoz) [inline frontend/src/api/metrics/getErrorPercentage.ts:2]: can we use single api as we are just passing query in each API

Comment (palash-signoz) [inline frontend/src/api/metrics/getErrorPercentage.ts:2]: can we use single api as we are just passing query in each API

Comment (aryanshridhar) [reply frontend/src/api/metrics/getErrorPercentage.ts:2]: Cool will do it!
How about we rename it to file to `getQueryEndpoint.ts` ?

File: frontend/src/container/MetricsApplication/Tabs/Application.tsx [hunk 2]
```diff
@@ -177,19 +179,33 @@ const Application = ({ getWidget }: DashboardProps): JSX.Element => {
...
-								noDataGraph
-								fullViewOptions={false}
-								onClickHandler={(event, element, chart, data): void => {
-									onClickhandler(event, element, chart, data, 'Request');
+								type="line"
+								data={{
+									datasets: applicationRpsEndpoints.map((data, index) => {
+										return {
+											data: data.values.map((value) =>
+												Number(parseFloat(value[1]).toFixed(2)),
```
Comment (palash-signoz) [inline frontend/src/container/MetricsApplication/Tabs/Application.tsx:192]: don't use to fixed here

File: frontend/src/container/MetricsApplication/Tabs/DBCall.tsx [hunk 3]
```diff
@@ -16,16 +21,26 @@ const DBCall = ({ getWidget }: DBCallProps): JSX.Element => {
...
-								fullViewOptions={false}
-								widget={getWidget([
-									{
-										query: `sum(rate(signoz_db_latency_count{service_name="${servicename}"}[1m])) by (db_system)`,
-										legend: '{{db_system}}',
-									},
-								])}
+								type="line"
+								data={{
+									datasets: dbRpsEndpoints.map((data, index) => {
```
Comment (palash-signoz) [inline frontend/src/container/MetricsApplication/Tabs/DBCall.tsx:28]: can we make a component component in which passing the state would reflect the same graph data

File: frontend/src/lib/getPayloadData.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import {
+	PayloadProps,
+	queryEndpointData,
```
Comment (palash-signoz) [inline frontend/src/lib/getPayloadData.ts:3]: getPayloadData is too generic can we make more specific ?

File: frontend/src/store/actions/metrics/getApplicationMetrics.ts [hunk 5]
```diff
@@ -1,27 +1,22 @@
...
-export const GetInitialData = (
+export const GetApplicationData = (
 	props: GetInitialDataProps,
 ): ((dispatch: Dispatch<AppActions>, getState: () => AppState) => void) => {
 	return async (dispatch, getState): Promise<void> => {
 		try {
 			const { globalTime } = getState();
 
-			/**
-			 * @description This is because we keeping the store as source of truth
```
Comment (palash-signoz) [inline frontend/src/store/actions/metrics/getApplicationMetrics.ts:23]: don't remove this one

File: frontend/src/store/actions/metrics/getApplicationMetrics.ts [hunk 6]
```diff
@@ -38,28 +33,17 @@ export const GetInitialData = (
 				globalTime.maxTime / 1000000,
 			]);
 
+			const start = getMicroSeconds({ time: minTime / 1000000 });
```
Comment (palash-signoz) [inline frontend/src/store/actions/metrics/getApplicationMetrics.ts:36]: use dayjs here

File: frontend/src/types/reducer/metrics.ts [hunk 7]
```diff
@@ -12,11 +9,15 @@ interface MetricReducer {
...
 	errorMessage: string;
-	dbOverView: DBOverView[];
-	externalService: ExternalService[];
 	topEndPoints: TopEndPoints[];
-	externalAverageDuration: ExternalAverageDuration[];
-	externalError: ExternalError[];
+	applicationRpsEndpoints: queryEndpointData[];
+	applicationErrorEndpoints: queryEndpointData[];
+	dbRpsEndpoints: queryEndpointData[];
+	dbAvgDurationEndpoints: queryEndpointData[];
```
Comment (palash-signoz) [inline frontend/src/types/reducer/metrics.ts:16]: can we add 3 tab level loading for each tab

**Review by palash-signoz — CHANGES_REQUESTED** [review]

File: frontend/src/store/actions/metrics/getApplicationMetrics.ts (hunk 5 above)
Comment (aryanshridhar) [reply frontend/src/store/actions/metrics/getApplicationMetrics.ts:23]: Oops!

File: frontend/src/store/actions/metrics/getApplicationMetrics.ts (hunk 6 above)
Comment (aryanshridhar) [reply frontend/src/store/actions/metrics/getApplicationMetrics.ts:36]: Not sure how dayjs will give the desired value.

**Comment by aryanshridhar** [comment]
Created a new pr for this - #859 and closing this one due to messed up commits while rebasing.

File: frontend/src/types/reducer/metrics.ts (hunk 7 above)
Comment (aryanshridhar) [reply frontend/src/types/reducer/metrics.ts:16]: Will address this issue in the new pr. Looking into it 👍

**Comment by aryanshridhar** [comment]
Note for self: Don't delete the branch.

**Pull request closed without merging**